tests/test_queries.py

```diff
@@ -0,0 +1,890 @@
+                logger.error(
+                    "Mismatch in electricity_cooling: actual=%s expected=%s",
+                    total_cooling,
+                    expected["electricity_cooling"],
+                )
+                success = False
+            if not math.isclose(total_heating, expected["electricity_heating"]):
+                logger.error(
+                    "Mismatch in electricity_heating: actual=%s expected=%s",
+                    total_heating,
+                    expected["electricity_heating"],
+                )
+                success = False
+        return success
+
+
+class QueryTestElectricityUseByCounty(QueryTestBase):
+
+    NAME = "electricity-use-by-county"
+
+    def make_query(self):
+        self._model = ProjectQueryResultModel(
+            name=self.NAME,
+            project=ProjectConstraintsModel(
+                project_id="dsgrid_conus_2022",
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                metric_reductions=[
+                    MetricReductionModel(dimension_query_name="electricity", operation="sum")
+                ],
+            ),
+            aggregations=[
+                AggregationModel(
+                    group_by_columns=["county"],
+                    aggregation_function=F.sum,
+                    name="sum",
+                ),
+                AggregationModel(
+                    group_by_columns=["county"],
+                    aggregation_function=F.max,
+                    name="max",
+                ),
+            ],
+            output_format="csv",
+        )
+        return self._model
+
+    def validate(self):
+        return validate_electricity_use_by_county(
+            self.output_dir / self.name / "sum" / "table.csv",
+            self.output_dir / self.name / "max" / "table.csv",
+            self.get_raw_stats(),
+        )
+
+
+class QueryTestElectricityUseByState(QueryTestBase):
+
+    NAME = "max-electricity-use-by-state"
+
+    def make_query(self):
+        self._model = ProjectQueryResultModel(
+            name=self.NAME,
+            project=ProjectConstraintsModel(
+                project_id="dsgrid_conus_2022",
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                metric_reductions=[
+                    MetricReductionModel(dimension_query_name="electricity", operation="max")
+                ],
+            ),
+            aggregations=[
+                AggregationModel(
+                    group_by_columns=["state"],
```

Comment:
Does this use the state county to state dimension mapping? Or does it just use the state field in the county dimension records? Do we have an example of a query that uses a dimension mapping?

Reply:
`group_by_columns` is a list of query names. If the query_name does not exist, it will fail. The query_name can be either a base dimension or a supplemental dimension. If it is a supplemental dimension we will look up the dimension mapping and apply it.